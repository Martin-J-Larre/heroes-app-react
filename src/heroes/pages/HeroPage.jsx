import React from "react";
import { Navigate, useNavigate, useParams } from "react-router-dom";
import { getHeroById } from "../helpers";

export const HeroPage = () => {
  const { id } = useParams();
  const navigate = useNavigate();

  const hero = getHeroById(id);

  if (!hero) {
    return <Navigate to="/marvel" />;
  }

  const heroImg = `/assets/heroes/heroes/${id}.jpg`;

  const onNavigateBack = () => {
    if (hero.publisher === "Marvel Comics") {
      navigate("/marvel");
    } else {
      navigate("/dc");
    }
  };

  return (
    <div className="row mt-5">
      <div className="col-4">
        <img className="img-thumbnail" src={heroImg} alt={hero.superhero} />
      </div>
      <div className="col-8">
        <h3>{hero.superhero}</h3>
        <ul className="list-group list-group-flush">
          <li className="list-group-item">
            <b>Alter ego:</b> {hero.alter_ego}
          </li>
          <li className="list-group-item">
            <b>Publisher:</b> {hero.publisher}
          </li>
          <li className="list-group-item">
            <b>First appearance:</b> {hero.first_appearance}
          </li>
        </ul>
        <h5 className="mt-3">Characters</h5>
        <p>{hero.characters}</p>

        <button className="btn btn-outline-dark" onClick={onNavigateBack}>
          Back
        </button>
      </div>
    </div>
  );
};
