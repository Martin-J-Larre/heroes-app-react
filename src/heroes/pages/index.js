export * from "./DcPage";
export * from "./Marvel";
export * from "./Search";
export * from "./HeroPage";
