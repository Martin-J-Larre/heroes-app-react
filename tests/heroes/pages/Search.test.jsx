import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { Search } from "../../../src/heroes/pages/Search";

describe("Test on Search.jsx", () => {
  test("should show component with defaul values", () => {
    const { container } = render(
      <MemoryRouter>
        <Search />
      </MemoryRouter>
    );

    // screen.debug();
    expect(container).toMatchSnapshot();
  });

  test("should show batman and the imput with queryString value", () => {
    render(
      <MemoryRouter initialEntries={["/search?q=batman"]}>
        <Search />
      </MemoryRouter>
    );

    const input = screen.getByRole("textbox");
    expect(input.value).toBe("batman");

    const img = screen.getByRole("img");
    expect(img.src).toContain("/heroes/dc-batman.jpg");

    const divDisplayNone = screen.getByLabelText("test-div");
    expect(divDisplayNone.style.display).toBe("none");

    // screen.debug();
  });
});
