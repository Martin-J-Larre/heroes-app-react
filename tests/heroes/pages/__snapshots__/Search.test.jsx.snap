// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Search.jsx should show component with defaul values 1`] = `
<div>
  <div
    class="row"
  >
    <h1>
      Search
    </h1>
    <hr />
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-primary mt-2"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Result
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        aria-label="test-div"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        There's no results for 
        <b>
          "
          "
        </b>
      </div>
    </div>
  </div>
</div>
`;
